File: src/main/java/com/group/libraryapp/domain/user/loanhistory/UserLoanHistoryRepository.java
package com.group.libraryapp.domain.user.loanhistory;

import com.group.libraryapp.user.loanhistory.UserLoanHistory;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserLoanHistoryRepository extends JpaRepository<UserLoanHistory, Long> {

  UserLoanHistory findByBookNameAndIsReturn(String bookName, boolean isReturn);

}

File: src/main/kotlin/com/group/libraryapp/domain/user/loanhistory/UserLoanHistoryRepository.kt
package com.group.libraryapp.domain.user.loanhistory

import org.springframework.data.jpa.repository.JpaRepository

interface UserLoanHistoryRepository : JpaRepository<UserLoanHistory, Long> {

//    querydsl로 변경
//    fun findByBookName(bookName: String): UserLoanHistory?
//
//    fun findByBookNameAndStatus(bookName: String, status: UserLoanStatus): UserLoanHistory?
//
//    fun countByStatus(status: UserLoanStatus): Long
}
